File: 0203_remove-linked-list-elements/solution.go
package _203_remove_linked_list_elements

type ListNode struct {
	Val  int
	Next *ListNode
}

func removeElements(head *ListNode, val int) *ListNode {
    for head != nil && head.Val == val {
        head = head.Next
    }
    p := head
    for p != nil && p.Next != nil {
        if p.Next.Val == val {
            p.Next = p.Next.Next
        } else {
            p = p.Next
        }
    }
    return head
}

File: 0203_remove-linked-list-elements/solution.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        dummy = ListNode(-1)
        dummy.next = head
        pre = dummy
        while pre.next:
            cur = pre.next
            if cur.val == val:
                pre.next = cur.next
            else:
                pre = pre.next
        return dummy.next
